clar2wasm/src/lib.rs

```diff
@@ -97,6 +99,46 @@ pub fn compile(
         }
     };
 
+    for expr in ast.expressions.iter() {
+        if expr
+            .match_list()
+            .and_then(|l| l.first())
+            .and_then(|first| first.match_atom())
+            .map(|atom| atom.as_str())
+            != Some("filter")
+        {
+            continue;
+        }
+
+        let Some(func_name) = expr.match_list().and_then(|l| l[1].match_atom()) else {
+            continue;
+        };
+
+        let entry_type = match contract_analysis.get_private_function(func_name.as_str()) {
```

Comment:
Is that a Clarity requirement that the function in a filter can only be private?

Reply:
public functions have to return responses, and thus cannot be used like a predicate function

Can you elaborate on what you mean by define-read-only?

Reply:
If you take the example in the `foo` test and replace the `define-private` by `define-read-only`, it should work (it works in the interpreter).

Reply:
are you sure? i cannot reproduce

Reply:
solved here https://github.com/stacks-network/clarity-wasm/pull/466#discussion_r1721971499